### ISSUE TITLE
Workshop Feedback

### ISSUE THREAD

**Issue by hexylena** [galaxyproject/training-material#1452]
The workshop feedbacks are getting buried a bit in the tutorial feedback due to volume, so I'm separating those out into this new issue :) Feel free to share your feedback from your workshop here, hopefully we can use this data to help identify common "papercuts" that new users are experiencing across workshops.

**Comment by erasche** [comment]
**Topics**: 
- Day 1: Intro, QC, Mapping (@erasche)
- Day 2: Annotation with MAKER (@abretaud)
- Day 3: 16S with Mothur (@shiltemann)

**Audience**: Fungal annotation people

**What did you like?**

Day 1:
- Easy to follow, nice atmosphere
- Good intro for someone with very little knowledge
- Material used was good/clear/easy to follow
- I like galaxy as a package no doubt, especially after command line in Linux, this is a relief
- "clean" information
- Slides are easy to read and process (not too many words per slide)

Day 2:
- More complex assignment, very straightforward
- great intro to annotation

Day 3:
- Really well explained
- Lots of resources
- Super convenient to design workflows
- Great introduction and also resource for general steps and procedures
- Was very nice to know about these tools from galaxy, today I enjoy more because it is more related to my project
- Thanks, I'll start using Galaxy much more in the coming weeks, months, years
- Very well done! Thank you very much!
- Overall a good course, provided much hope and leads!!


**What could be improved?**
Day 1:

- Could you please point to cove genome trees / snp based trees
- Please spend more time on parameter selection / critera / instrument application / optimum reasoning behind executing certain steps
- We had quite a bit of spare time, without reading a lot into what/how each tool does, which parameters it uses, etc. But galaxy is pretty strightforward
- I missed a bit more explanation about why we do these steps, more than how to do them. Also interpretation of the outputs
- Lot of information in a short time, but that is something that can't be avoided
- Improvement, more information on "pitfalls", e.g. mistakes people make when running analyses
- More info about diff parameters
- Longer introduction on: errors with examples (QC), different tools for diff sequencing techniques
- Explain in detail the different tools, specially those that are not familiar with the program/genomics
- How to extract for methods section

Day 2:

- For galaxy dev, would it be possible to group items in the history? That would enable collapsing reports or part that will become less relevant, e.g. group all the QC steps (responded with: collections the next day)
- Version bug in tutorial (@erasche fixed on .eu)
- Including assembly before annotation would be great
- less repetitive tutorials, e.g. same tools used too many times

Day 3:

- More varied tutorials / less repetitive
- Fungal versions would be nice (but understand due to the "numbers game")
- Would be nice if there is a little more background of the subject before start with all the tools.
- Although the overall topics are a lot, I would like to have a dedicated course on nanopore

_Originally posted by @erasche in https://github.com/galaxyproject/training-material/issues/989#issuecomment-493438843_

**Comment by erasche** [comment]
@erasche great idea!

Here's our workshop feedback from last week.

**Intro to Galaxy and RNA-seq** 
2 x 3hr sessions
[Material used](https://pmacdasci.github.io/galaxy-intro-2019-05-06/) 
(tried this format for first time, previously tried [4 x 2hr](https://pmacdasci.github.io/galaxy-intro/))

**Audience**: Cancer biologists

**What did you like about this workshop?**

* Hands on and approachable
* it is practical
* Really well presented and explained, extra resources (pdfs, training material) very helpful
* Day 2!! Day 1 was a disaster for me as it went too quick and I wasn't prepared, and had a few minor technical issues. Overall the content and content level was appropriate
* I liked that fact that some theories was explained behind the data representation and what all the Figures mean and implying it to. I like that it was explained in very detail and step by step so that it could be followed.
* Hand on Experience
* Very clear explanation of theory and application methods. Well supported by the team when problems are encountered.
* Very direct tutorial of the actual tools we would use, rather than just generic discussion

**What could be improved?**

* Help understand some of the data
* More training courses are needed
* Although we already covered quite a lot in a short amount of time, would be good if we had more time to discuss the different tools and different options for different types of data in more detail. This would probably require more sessions
* Difficult to follow at the beginning.
* Will be better if you can send out the course materials earlier.
* Pre-reading the Galaxy Introduction document should be suggested to the next group
* The number of sessions should be more. In particular the pathway analysis could have been shown in much more detail with the increased number of sessions.
* Can include more analysis
* Perhaps reducing speed when processing the more complex demonstration data. As with many technical courses, things moving too fast can make people fall behind and become frustrated. I don't see an easy way around this though as it is all important and needed.
* If you get people to read/familiarise themselves with some basic principals/terms first (ahead of the course) it would save time (P-value/multiple testing corrections, GO terms).

**Any further comments or suggestions?**

* Get Galaxy to improve their interface! For some actions its impossible to know what to do
* Yes, the future workshops may be considered running as multiple short sessions. It would have been great to give an option to researchers to come along in future workshops just for the day they are interested to learn. In that way we can only attend the part we are not familiar with or we are interested on. It also would be helpful for the lab scientists since it is hard to find time to learn new bioinformatics skill.

**Myself and helper's (@rlupat) thoughts on improving**

*General*

* Send some material (at least the Introduction tutorial) before the course as pre-reading
* Try sessions of 2hr with additional optional 1 hr Q&A for people to hang around to ask questions, get more details, ask about own data
* Create histories and share with user emails in advance. As trying to share histories during the session to help people catch up caused us issues during the course, as tried initially sharing histories by link but then any new datasets created were locked to users (permissions) so had to reshare, share individually with emails
* See if can get a mouse highlighter app to make more obvious where the cursor is pointing (suggested by attendee) e.g. https://itunes.apple.com/us/app/mypoint-light-s/id518658487?mt=12

*Material*

Introduction
* Send out in advance

Reads to Counts
* Only run one MultiQC to demo as too confusing when have multiple MultiQCs in history, they all look the same
* Focus on the main steps Trim > Align > Count (and then show info/workflow for QCs)
* Maybe get people to load in data from Shared History instead of Rule Builder to make more straightforward (put Rule Builder method in More details box?) as the majority were new users just getting used to Galaxy. As then they'd be no need to strip header, assign columns, build etc, they could just start working with the data. Also may avoid some issues, as if data is pre-loaded could have datatype and genome preselected, as some users selected wrong datatype (fastqcsanger) or genome (mm10Strains1).

Counts to genes material
* Drop the beginning, formatting data, part (or put in a More details box) and provide the data already formatted so all they have to do is run the annotatemyids tools and limma, as it takes a while to run that part and they could (and likely will) format in Excel.
* Add in part on batch effects

Genes to pathways 
* Needed more time, was too squished in at the end. Go back to running as a separate module.


*UI Feedback*

* If sharing histories with email address it works ok, if sharing by link, need to make sure to tick to make the datasets accessible so people don't get permissions errors when the try to run. But then if run more tools the new datasets are locked so have to unshare and reshare the link (or share with emails). That caused a lot of confusion!
* Multiple people tried to rename a history using the history search box which then caused them confusion because it filtered their history datasets and they couldn't see any data in their history. One user said they tried to rename using the search box because "Unnamed history" did not look editable (we had covered renaming histories but it was a lot of new info for them to remember). If you mouse over "Unamed history" it does say "Click to rename history" but they had already made the assumption that it was the search box that was the history name so the mouse over didn't help. If it was possible to make it more obvious that "Unamed history" is editable that could be good.
* Having something like an "Import" button for importing histories would be good if possible, instead of the "+" button that's over on the right side when viewing a shared history, as it wasn't obvious to them that + was the way to import.
* The rerun button on collections was reported as hard to find
* A user asked if there was a way to view a history like in the workflow editor e.g. so can see what steps are orphan? Said yes would be great but not possible at the moment.
* Having a way to edit bulk edit datatypes and database would be useful for users that make mistakes selecting those when importing data, will try to work around that by pre-loading data into Shared History. Btw this is one reason we ran into the history sharing issues as tried to share histories with users who had made errors with importing but who didn't realise til after they had run a few tools on the data.

_Originally posted by @mblue9 in https://github.com/galaxyproject/training-material/issues/989#issuecomment-493642457

**Comment by erasche** [comment]
**Intro to Galaxy Administration** at GCC2019

**Topics**:

- Welcome (@martenson)
- Ansible (@Slugger70)
- Ansible for Galaxy installation (@erasche)
- Connecting to a compute cluster (@natefoo)

Helpers: @jmchilton

**Audience**: Administrators (from beginning to advanced) and some developers

**What did you like about this workshop?**

- Very nice, thanks for help
- Post it green/red it's a really nice idea!
- awesome material and presenters. I *learned a lot* today
- Nice to have a wide overview of so many aspects of sysadmin with galaxy. Thanks!!
- Help was reactive, thans!
- Thanks for teaching ansible and galaxy cluster interface
- On the fly solutions of problems
- adaptability to the audience
- topics covered
- good structure and nice tutorial hands on
- great instructors, friendly environment
- great help/team
- feedback system, proactive helpers
- Switching to immediate best practices instead of "simplest setup + improve" like previous years is A++! Thanks for your work!
- Serving the live files +1 So that we can get caught up + diff to quickly find errors
- Ansible playbook was a productive exercise what I've need all these years!
- Useful and descriptive tutorials with many clear examples
- Great job everyone, liked gitter channels and self-assigning machines at start
- You were really helpful and there were a good number  oof helpers
- Text pages are good
- Well organised/strutured teaching with optimal pace
- Great organisation / very helpful / very good presentations

**What could be improved?**

- Just a quick thought about restructuring for 1 day only, skip the incremental "build galaxy instance with ansible" and just run the full thing immediately. Then spend more time on job config/etc.
- Should know ansible as prerequisite and drop it.
- Before breaks, all machines should be aligned + working (so there is time to fix problems)
- Would have been nice to get 5 min to synthesize(?) command line at the end to fix ideas because many new motions + concept.
- Not all modifications of files are clearly defined
- It was faster during copy in different files
- When you get stuck it can be really challenging to catch up because content overload
- [3x] Providing working + completed playbooks early on
- Too intensive for one day
"- A lot of material, didn't get a chance to finish
- A bit fast and couldn't listen to the "tips"

*Known bugs*

- My SSH connection was blocked several times and made me lose time
- At a few important steps, the attempted file/directory is missing (unsure of rest of comment)
- [5x] VM was too small

*Out of scope*

- Need a little more info regarding optimising galaxy for production, e.g. diff threads/processes/handlers
- Covering *ephemeris* (but it's only a 1 day workshop, guess easier over 5 days!)
- Taking specific examples from a chem-bio theme will be even better :)


**Our thoughts on improving**

- #1445 
  - swap to staging environment for certbot
  - Be clear about ordering of roles
- Remove some of the "conversational style and just be explicit about what goes where
- Specify in the requirements of the tuto, the minimum VM size.
- Add a box explaining role dependencies
- Add a box explaining templating when first used (i.e. 'at runtime' stuff.)
- Add welcome page in nginx route
- One true ansible setup (on remote machine + desribe why bad in prod) xref https://github.com/galaxyproject/training-material/issues/1436
- Limit apocalypse to not affect client build maybe?

**Comment by erasche** [comment]
Galaxy Intro at MGC2019 at LUMC

**Topics**:

- Galaxy 101 (@shiltemann @erasche)

**Audience**: Everyone from wet lab to bioinformatics students

**What did you like about this workshop?**

We informally collected feedback, all positive, useful for wet lab people even.

**Notes / Bugs**

- [ ] The "rename dataset" in the workflow editor "saves" when leaving the field, and users found this strange after "enter to save" in the history title. We should just unify this to "save when exiting the field / auto-save even when typing"
- [ ] We need a slide on training materials in the gx intro
- [ ] Not clear to expand the dataset for lines, should mention this in the tuto
- [ ] EU has two sort tools, diff versions, in diff sections. We should fix this.
- [ ] Minimap is no longer colour coded, so we should remove this image.
- [ ] The workflow layout is "bad" in the extracted from history step, many users had questions over "it doesn't have an input from the step one", we should encourage rearranging to see connections due to overlap.
- [ ] UCSC step to fetch data in "run WF on diff data" step does not say "run the UCSC tool", just "obtain from UCSC".

**Comment by erasche** [comment]
Volcano plots with R, popup workshop at INAB|CERTH

**Topics**
- RNA Seq Counts to Viz in R  (@erasche) 

**Audience**: ca 10 bioinformaticians and programmers, all had some experience with R, a couple with Galaxy

**What did they like**
(Feedback collected informally)

**Bugs / TODOs**
- [ ] add tip box about getting help for R functions (as a reminder, and for those who didn't do basic R tuto before this one)
- [ ] Add links to ggplot documentation or tutorials
- [ ] explain what mapping, aes, etc are with  better words than code block. they're copy pasting the first block, make it more clear it's explanation
  - series of images building up the plot with different layers?
  - show overview at beginning
- [ ] make question box about "130 observations of 13 variables" to check for errors in importing

- [ ] More bio explanation on "why these plots", or links to that sort of thing in details boxes.

**User Questions**
- [ ] Q: explain alpha variable for transparency (why are some getting it)
  A: all are transparent but due to overlap some seem opaque
- [ ] Q: why is color variable first used in geom.point, and then used in ggplot?
  A: parameters like color, shape, fill are general and can be applicable to different layers. If you want this to be applied to a certain variable of data frame, (geom.point will take precedence, higher layer). Parameters in aesthetics can be connected directly to your data field
- [ ] Q: why do we need the standard error plot?
  A: not per say needed, depends on what you want to do, but may provide useful information in some cases. In this example chromosome X shows ..
- [ ] Q: how do we plot everything but color everything from one chromosome differently? Or highlight a list of genes of interest?
  A: define colour scheme manually, or split plot by chromosome. there are ways to do this
- [ ] Ideas for steps to add to plotting:
  - highlighting points by some property, e.g. colour chrX differently than rest.
  - Add chromosome labels (possibly from a llist of genes of interest)
  - colour by coding/noncoding
- [ ] Q: how can we make a venn diagram?
  A: there are options in ggplot, also a Galaxy tool for it.

**Comment by hexylena** [comment]
**Galaxy Admin Training** at the Barcelona Supercomputer Center

**Topics:**

- all of admin basically (@shiltemann @hexylena @natefoo @nsoranzo @mvdbeek)

**Audience:** Everyone seemed super competent with command line already? (1 emacs user, 6 nano, 20+ vim). All galaxy admins.

**What did you like about this workshop?**

typos are my own.

*day 3*

- practices are useful
- documenting questions
- well organised
- total positive attitude of teacher :)
- very good organisation with stickers
- exercise tutorils are exhaustive
- trainer's availability is appreciated
- enjoyed the adition of graphical reporting
- transformative!
- great teachers and slides
- exciting
- enjoyed cvmfs
- you can learn details about galaxy but also other useful unix tools!
- really like you are addressing these aspects of galaxy
- learning so many things

Cons:
- heavy workload
- too much info to absorb
- try to avoid people being helped + continuing the presentation (No, sorry, this - is by design, won't fix..)
- [ ] hard to focus on compute cluster slides, too much text
- only once a year?
- maybe we need something like a summary table (addressed)
- still a bit fast
- not clear what needs to be where in ansible and it seems crucial
- at the beginning it was difficult sometimes to follow which files to modify
- as I understnad more it's not that hard

*day 1*

Cons:

- maybe fast in some ansible sections
- would have liked more time on ansible, got very complex v quickly, build up more slowly?
- following some "just" code slides could be difficult
- show commands on screen more than once
- galaxy is a veyr complex piece of software and we seem to have rushed through a bit with the explanations (you should know what galaxy is already?)
- speed x 4

Pros:
- almost everything, vms ready to use, google doc for Qs, presentations
- well organised
- good red yellow stickers are very useful
- exhaustive
- lot of new things learnt
- great system with VMs / presentations
- instructors, environment, materials (despite ports in vm training room)
- quality of online material is excellent
- pace
- good pace and clear materials
- friendly and willing to help
- individual help
- tours
- perfect so far

**Conclusions**

These will be discussed in #1809 but we have some ideas to improve the educational aspects.

**Comment by Slugger70** [comment]
**Galaxy 101 for everyone**

The University of Melbourne, Australia - 24th & 25th September 2020 (Two sessions)

Audience: Various science personnel - mostly post grads. Conducted online. 

Method: Online

Likes: Really good general introduction to Galaxy and workflows. Lots of positive feedback from the students. Was very easy to teach.

Cons: We were using Galaxy 20.05 and some of the screenshots and instructions are out of date 😞 . Not too big a deal as facilitators and trainers can easily point out differences.

Conclusion: A really excellent tutorial.

**Comment by igormakunin** [comment]
De novo transcriptome reconstruction with RNA-Seq

The University of Queensland, Australia - 13th October 2020 (one session). 
Audience: mix, including researchers from other institutions. 

Method: Online (zoom)

Pros: Well documented, uses popular tools (HiSAT2, StringTie, featureCounts). 

Cons: No explanation for advanced settings used with HiSAT2 esp for spliced alignment. 

It seems there is minor typo: HiSAT2 uses mm10, while in the Visualization step bamCoverage uses mm9 genome size.

Also, Megakaryocyte_rep2 files have under 100k reads, while the remaining three samples have ~1M. Not sure if it is done on purpose, but very useful for QC. 

Very positive impression from the tutorial.

**Comment by fpsom** [comment]
[**Reference-based RNA-Seq data analysis**](https://training.galaxyproject.org/training-material/topics/transcriptomics/tutorials/ref-based/tutorial.html)
- **Context**: invited lecture, [graduate program on Precision Medicine](http://precision.med.auth.gr/), School of Medicine, Aristotle University of Thessaloniki
- **Date**: November 14th, 2020
- **Delivery method**: fully virtual tutorial (zoom)
- **Duration**: ~5 hours (reached the ["filtering p-adj value" section](https://training.galaxyproject.org/training-material/topics/transcriptomics/tutorials/ref-based/tutorial.html#extraction-and-annotation-of-differentially-expressed-genes))
- **Audience**: Mostly graduate medical students, a few clinicians

General feedback (participants' perspective)
1. Appear to enjoy the detailed instructions for each step. Helped clarify points in QA, mapping.
2. Most questions were related to the clinical usefulness of each result (esp. after reaching the DE section)
3. Would like to have more disease-specific scenarios (i.e. tailor the input dataset / overall questions to match their own interests).

_I will update this entry, if additional feedback is offered_

Feedback (instructors' perspective)
I believe that this particular lesson is well-suited for a short tutorial, even in a traditional academic context. The first 2.5 hours were dedicated to the QA/Mapping/Counting, and the rest ~2 hours were for explaining how DE works. 
The dedicated training queue for executing jobs worked rather well (incl. the interface that allowed for the monitoring of jobs in real-time).

**Comment by burkemlou** [comment]
**Workshop Name:** Online data analysis for biologists (a.k.a Galaxy 101 for everyone)

**Date:** 12 November 2020

**URL:** https://www.biocommons.org.au/events/galaxynov2020

**Format:** 3 hrs. Online via Zoom using a combination of lecture by the lead trainer and breakout rooms with facilitators where trainees worked through the exercises.

**Materials used:** Galaxy 101 for everyone https://training.galaxyproject.org/training-material/topics/introduction/tutorials/galaxy-intro-101-everyone/tutorial.html

**Number of participants:** 17 trainees, 5 trainers

**Audience:** Biologists with no prior knowledge of Galaxy and no programming experience.

**What did you like about this workshop?** 

- The materials can be used in multiple contexts and by different trainers to deliver consistent content. New trainers can quickly become familiar with the content by working through the tutorial themselves.

- The materials use a biological example to demonstrate how to do simple tasks in Galaxy. This helps participants to imagine how they might use it with their own research.

- Allocation of training servers on Galaxy Europe allowed us to run this workshop while Galaxy Australia was undergoing maintenance. We appreciate that this was set up quickly and at short notice - thank you!


**What could be improved?**

- The written tutorial doesn't tell you what to do if there is more than one version of a tool. Different versions of tools give different results and have different options when setting up the analysis pipelines which can be confusing for participants.

- Tool search froze occasionally 

**Any further comments/suggestions**

- There is not a lot of time in the 3 hour schedule for troubleshooting. Trainers need to be aware that there is no time to debug.  Instead use shared histories and workflows (prepared ahead of time) to allow the tutorial to be continued.  

- Interaction with participants was variable. In future versions of this workshop we will try to encourage more interaction between participants and trainers e.g. by more clearly setting expectations, adjusting trainer:participant ratios in breakout rooms, encouraging group work.

**Comment by hexylena** [comment]
> The written tutorial doesn't tell you what to do if there is more than one version of a tool. Different versions of tools give different results and have different options when setting up the analysis pipelines which can be confusing for participants.

@burkemlou be sure to check out the GTN-in-Galaxy feature, it addresses this issue, at least on EU :) https://training.galaxyproject.org/training-material/topics/contributing/tutorials/create-new-tutorial-content/tutorial.html#tool-links
